File: Backend/OnnorokomBackend/Models/Entities/Assignment.cs
using OnnoRokomBackend.Models.Enums;

namespace OnnoRokomBackend.Models.Entities;

public class Assignment
{
    public Guid Id { get; set; }
    public Guid CourseId { get; set; }
    public Guid CreatedByUserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime DeadlineAt { get; set; }
    public decimal MaximumMarks { get; set; }
    public AssignmentStatus Status { get; set; } = AssignmentStatus.Draft;
    public bool AllowResubmission { get; set; }
    public DateTime? SubmissionsClosedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public Course Course { get; set; } = null!;
    public User CreatedBy { get; set; } = null!;
    public ICollection<Submission> Submissions { get; set; } = [];
}

File: Frontend/src/types/assignment.ts
import type { AssignmentStatus } from "./enums";

export interface AssignmentResponse {
  id: string;
  courseId: string;
  courseCode: string;
  courseTitle: string;
  title: string;
  description: string | null;
  deadlineAt: string;
  maximumMarks: number;
  status: AssignmentStatus;
  allowResubmission: boolean;
  submissionsClosedAt: string | null;
  createdByUserId: string;
  createdByName: string;
}

export interface CreateAssignmentRequest {
  courseId: string;
  title: string;
  description?: string | null;
  deadlineAt: string;
  maximumMarks: number;
  allowResubmission: boolean;
}

export interface UpdateAssignmentRequest {
  title: string;
  description?: string | null;
  deadlineAt: string;
  maximumMarks: number;
  allowResubmission: boolean;
}
